dev/Code/CryEngine/CryAnimation/ModelMesh.cpp

```diff
@@ -171,7 +171,7 @@ ClosestTri CModelMesh::GetAttachmentTriangle(const Vec3& RMWPosition, const Join
         Vec3 v1 = *(Vec3*)(pPositions + pIndices[d + 1] * nPositionStride);
         Vec3 v2 = *(Vec3*)(pPositions + pIndices[d + 2] * nPositionStride);
         Vec3 TriMiddle = (v0 + v1 + v2) / 3.0f + m_vRenderMeshOffset;
-        f32 sqdist = (TriMiddle - RMWPosition) | (TriMiddle - RMWPosition);
+        f32 sqdist = (TriMiddle - RMWPosition) | (TriMiddle - RMWPosition); //(TriMiddle - RMWPosition) duplicated, possible typo
```

Comment:
I have just been looking at this one and wondered if you had addressed it. It had me confused at first, but the code is actually correct. The `|` operator is overloaded to perform the dot product on two Vec3s. If the vectors are parallel then the dot product just results in the ordinary multiplication of their lengths. If the vectors were normalised then this would just result in 1, but as they are not we get the squared distance. It's not clear, but it is correct!

Reply:
Haha yeah, I am very much a fan of optimizing for the reader. The compiler is pretty smart whereas mortal programmers get confused easily! It took me longer than I am willing to admit to realise what that code was doing ;)